File: Day62/csp.js
int isSumProperty(Node *node)
{
 // Add your code here
    int ld=0,rd=0;
    if(!node || (!node->left && !node->right))return 1;//one will corresspond to the &&
    else{
        if(node->left)
            ld = node->left->data;
        if(node->right)
            rd = node->right->data;
        if(
            (node->data == (ld + rd))
                && isSumProperty(node->left)
                    && isSumProperty(node->right)
        
        )return 1;
        else return 0;
    }
 
}

File: Day216/csp.cpp
class Solution{
    public:
    //Function to check whether all nodes of a tree have the value 
    //equal to the sum of their child nodes.
    int isSumProperty(Node *root)
    {
        if(!root)return 1;
        if(!root->left && !root->right) return 1;
        
        int ls = root->left?root->left->data:0;
        int rs = root->right?root->right->data:0;
        
        return (root->data == ls+rs &&
                isSumProperty(root->left)
                && isSumProperty(root->right));
    }
};